proto/common.proto

```diff
@@ -25,15 +25,23 @@ message SpanId {
 }
 
 message Field {
-    oneof field {
-        string debug = 1;
+    oneof name {
+        string str_name = 1;
+        uint64 name_idx = 2;
```

Comment:
in order to actually use this, we'll also want to include the list of field names in the metadata message, so that these indices are meaningful. but we don't have to do that in this PR, that can be added in a follow-up.

Reply:
also, just a thought: it _might_ be worth having this message also include the metadata ID that the index refers to, so that a `Field` message with a name index can always be hydrated? but, since the `Field` message is only ever sent in the `Span` and `Task` messages, which also include metadata IDs, this isn't strictly _necessary_...